### DISCUSSION TITLE
Wait for server endless

### DISCUSSION THREAD

**Discussion by mago0211** [ipxe/ipxe#994] [Q&A] [answered]
Hi,
is there a way that I can set iPXE into a endless discovery loop for server.

I will try to build an automated test envrionment with 2 VM's. 1st VM the server and 2nd VM the Client(with PXE boot). In my case it is neccessary to restart the server and the client (e.g. Software update). But when the client starts and the server is not yet ready.Then the client give up the searching after (about 10 sec).  And I must press a key to restart the VM but this is for a automated test envionment not applicabel.

For me, the best thing would be to be able to tell him to wait indefinitely for a server. Or for a long time.

Thanks

**Comment by NiKiZe** [comment, marked as answer]
If network is "unreliable" you can always use embedded script and goto to create endless loop. See https://ipxe.org/scripting

Be aware tho, it might cause memory leaks or other resource leaks to surface that hangs. So after X number of failed attempts you might want to implement restart.

```
:theloop
# Todo, add counter and reboot
autoboot || goto theloop 
```

**Reply by mago0211** [reply]
Awesome. 🙏
Thank you for the quick and perfect reply.

(I'm a little sad that I didn't think of it myself.)
